differential-dataflow/src/operators/reduce.rs

```diff
@@ -533,7 +533,8 @@ where
                                 for (val, time, diff) in buffers[index].1.drain(..) {
                                     buffer.push_into(((T1::owned_key(key), val), time, diff));
                                     builders[index].push(&mut buffer);
-                                    buffer.clear();
+                                    // TODO: Clear!
+                                    // buffer.clear();
```

Comment:
This is a moment where we need to think a bit:
* We could add a constraint to `Input` that we can call `clear` in it. Easy, but also not forward-thinking.
* What we're doing here is building chunks from elements. We hope that the order of items in the chunk is the same as in `buffers`, otherwise we'd construct an invalid batch. Elsewhere, we'd use a container builder that feeds into a batcher, and the batcher's output is guaranteed to be suitable for the builder.
* We're also restricting the container to absorbing _owned_ data, but we might want references here.

Reply:
Changed to use `MergerChunk::clear` here. Not ideal, but works.